src/Glpi/Form/ServiceCatalog/Provider/CategoryProvider.php

```diff
@@ -87,4 +88,53 @@ public function getItems(ItemRequest $item_request): array
 
         return $categories;
     }
+
+    /**
+     * @param ItemRequest $item_request
+     * @param bool $return_current Include the current category in the result
+     * @return array<array{id: int, name: string}>
+     */
+    public function getAncestors(ItemRequest $item_request, bool $return_current = true): array
+    {
+        $category_id = $item_request->getCategoryID();
+        $category = Category::getById($category_id);
+        if (!$category) {
+            return [];
+        }
+
+        $categories = [];
+        $current_category = [
+            'id' => $category->getID(),
+            'name' => $category->fields['name'],
+        ];
+
+        $ancestors_iterator = $category->getAncestors();
+        if (!$ancestors_iterator) {
+            if ($return_current) {
+                return [$current_category];
+            }
+            return [];
+        }
+
+        /** @var Category[] $ancestors */
+        $ancestors = iterator_to_array($category->getAncestors());
+        foreach ($ancestors as $ancestor) {
+            // We check the ancestor has form linked to it
+            $count = countElementsInTable(Form::getTable(), ['forms_categories_id' => $ancestor->getID()]);
```

Comment:
This will add many requests. It should be passed to the `Category::getAncestors()` method to filter the result in the query used to get the list of ancestors.